pkg/e2etests/metrics_e2e_test.go

```diff
@@ -0,0 +1,1186 @@
+	assert.Nil(t, err)
+	assert.Equal(t, uint32(20), intervalSeconds)
+
+	query := `({__name__=~"testmetric.*"})`
+	metricQueryRequest, _, _, err := promql.ConvertPromQLToMetricsQuery(query, timeRange.StartEpochSec, timeRange.EndEpochSec, 0)
+	assert.Nil(t, err)
+
+	/*
+		Expected Results:
+		At T1: testmetric0: 10, testmetric1: 20, testmetric2: 30
+		At T2: testmetric0: 40, testmetric1: 80, testmetric2: 120
+		At T3: testmetric0: 50, testmetric1: 90, testmetric2: 130
+		At T4: testmetric0: 60, testmetric1: 100, testmetric2: 140
+		At T5: testmetric0: 70, testmetric1: 110, testmetric2: 150
+
+		The time diff between T1 and T2 is less than 6 seconds, so the values at T1 and T2 will be aggregated to the same bucket.
+
+		Values at T1 bucket = 10 + 20 + 30 + 40 + 80 + 120 = 300 / 6 = 50
+		Values at T2 bucket = 50 + 90 + 130  = 270 / 3 = 90
+		Values at T3 bucket = 60 + 100 + 140 = 300 / 3 = 100
+		Values at T4 bucket = 70 + 110 + 150 = 330 / 3 = 110
+	*/
+	expectedResults := []float64{50, 90, 100, 110}
+
+	res := segment.ExecuteMetricsQuery(&metricQueryRequest[0].MetricsQuery, &metricQueryRequest[0].TimeRange, 0)
+	assert.NotNil(t, res)
+	assert.Equal(t, 1, len(res.Results))
+
+	seriesDp := res.Results["*{"]
+	assert.NotNil(t, seriesDp)
+	assert.Greater(t, len(seriesDp), 0)
+
+	seriesDpValues := make([]float64, 0)
+	for _, dp := range seriesDp {
+		seriesDpValues = append(seriesDpValues, dp)
+	}
+	sort.Slice(seriesDpValues, func(i, j int) bool {
+		return seriesDpValues[i] < seriesDpValues[j]
+	})
+
+	assert.EqualValues(t, expectedResults, seriesDpValues)
+}
+
+func Test_SimpleMetricQuery_Regex_on_MetricName_OR(t *testing.T) {
+	defer cleanUp(t)
+
+	startTimestamp := dataStartTimestamp
+	allTimeSeries, _, _, _ := GetTestMetricsData(startTimestamp)
+
+	err := initTestConfig()
+	assert.Nil(t, err)
+
+	err = ingestTestMetricsData(allTimeSeries)
+	assert.Nil(t, err)
+
+	mSegs, err := rotateMetricsDataAndClearSegStore(true)
+	assert.Nil(t, err)
+	assert.Greater(t, len(mSegs), 0)
+
+	err = initializeMetricsMetaData()
+	assert.Nil(t, err)
+
+	timeRange := &dtypeutils.MetricsTimeRange{
+		StartEpochSec: uint32(startTimestamp),
+		EndEpochSec:   uint32(startTimestamp + 4600),
+	}
+
+	intervalSeconds, err := mresults.CalculateInterval(timeRange.EndEpochSec - timeRange.StartEpochSec)
+	assert.Nil(t, err)
+	assert.Equal(t, uint32(20), intervalSeconds)
+
+	query := `({__name__=~"testmetric(0|1)"})`
+	metricQueryRequest, _, _, err := promql.ConvertPromQLToMetricsQuery(query, timeRange.StartEpochSec, timeRange.EndEpochSec, 0)
+	assert.Nil(t, err)
+
+	/*
+		Expected Results:
+		At T1: testmetric0: 10, testmetric1: 20
+		At T2: testmetric0: 40, testmetric1: 80
+		At T3: testmetric0: 50, testmetric1: 90
+		At T4: testmetric0: 60, testmetric1: 100
+		At T5: testmetric0: 70, testmetric1: 110
+
+		The time diff between T1 and T2 is less than 5 seconds, so the values at T1 and T2 will be aggregated to the same bucket.
```

Comment:
Why is it 5 seconds in this test but 6 seconds in the above test? (I'm just missing where these numbers are coming from)

Reply:
These numbers are from data generation part above from the function `GetTestMetricsData`. I will update these comments. 

And I have written these comments as to explain myself and others how I have arrived at these expected results.